syntax = "proto3";

service StockTrading {
  rpc Lookup (LookupRequest) returns (LookupResponse);
  rpc Trade (TradeRequest) returns (TradeResponse);
  rpc Update (UpdateRequest) returns (UpdateResponse);
}

message LookupRequest {
  string stock_name = 1;
}

message LookupResponse {
  float price = 1;
  int32 trading_volume = 2;
}

message TradeRequest {
  string stock_name = 1;
  int32 quantity = 2;
  enum TradeType {
    BUY = 0;
    SELL = 1;
  }
  TradeType type = 3;
}

message TradeResponse {
  int32 status = 1;
}

message UpdateRequest {
  string stock_name = 1;
  float new_price = 2;
}

message UpdateResponse {
  int32 status = 1;
}
